react-components/src/architecture/concrete/observationsDomainObject/ObservationsDomainObject.ts

```diff
@@ -0,0 +1,68 @@
+/*!
+ * Copyright 2024 Cognite AS
+ */
+import { CDF_TO_VIEWER_TRANSFORMATION, Overlay3DCollection } from '@cognite/reveal';
+import { OBSERVATION_SOURCE, type ObservationProperties, type Observation } from './models';
+import { VisualDomainObject } from '../../base/domainObjects/VisualDomainObject';
+import { type ThreeView } from '../../base/views/ThreeView';
+import { ObservationsView } from './ObservationsView';
+import { type TranslateKey } from '../../base/utilities/TranslateKey';
+import { type FdmSDK, type InstanceFilter } from '../../../utilities/FdmSDK';
+import { Vector3 } from 'three';
+import { DEFAULT_OVERLAY_COLOR } from './constants';
+import { Changes } from '../../base/domainObjectsHelpers/Changes';
+
+export class ObservationsDomainObject extends VisualDomainObject {
+  private readonly _collection = new Overlay3DCollection<Observation>([], {
+    defaultOverlayColor: DEFAULT_OVERLAY_COLOR
+  });
+
+  public override get typeName(): TranslateKey {
+    return { fallback: ObservationsDomainObject.name };
+  }
+
+  constructor(fdmSdk: FdmSDK) {
+    super();
+    void fetchObservations(fdmSdk).then((observations) => {
+      this.initializeCollection(observations);
+      this.notify(Changes.geometry);
```

Comment:
Skip notify here. Should not be called in the constructor. Unnecessary, since constructor change everything.

Reply:
It's not called directly in the constructor, it's called as a reaction to a promise (in the `.then` function). I want to notify when `fetchObservations` is done, which is an async function